rq/job.py

```diff
@@ -485,6 +485,8 @@ def delete(self, pipeline=None):
     # Job execution
     def perform(self):  # noqa
         """Invokes the job function with the job arguments."""
+        self.connection.persist(self.key)
+        self.ttl = self.connection.ttl(self.key)
```

Comment:
Yeah, see https://github.com/nvie/rq/pull/477#issuecomment-76408703 we could hard-code it to `-1`